src/hooks/useBalanceChartData/useBalanceChartData.ts

```diff
@@ -260,8 +272,20 @@ const calculateBucketPrices: CalculateBucketPrices = (args): Bucket[] => {
           break
         }
         case TxType.Receive: {
-          // we're going backwards, so a receive means we had less before
-          bucket.balance.crypto[txAssetCAIP19] = bucket.balance.crypto[txAssetCAIP19].minus(value)
+          // for some contract interactions, the txAsset may be a token (e.g. erc20)
+          // but we may be on the fee asset (e.g. eth) chart
+          if (bucket.balance.crypto[txAssetCAIP19]) {
+            // we're going backwards, so a receive means we had less before
+            bucket.balance.crypto[txAssetCAIP19] = bucket.balance.crypto[txAssetCAIP19].minus(value)
+          }
+
+          // some txs, e.g. claim an airdrop, require gas but are receives of tokens
+          if (bucket.balance.crypto[feeAssetCAIP19]) {
+            // we're going backwards, so a send means we had more before
+            // and even though this is a receive tx of a token, we sent the fee
+            bucket.balance.crypto[feeAssetCAIP19] =
```

Comment:
Might we want to hoist this if block and feeValue addition out of the switch/case and do it for all transactions?